src/common/snippets/src/pass/split_dimension_m.cpp

```diff
@@ -42,6 +44,19 @@ std::pair<size_t, size_t> SplitDimensionM::get_splited_dimensions(size_t batch_d
         return splited;
     }
 
+    // Ideal case #2: M is divisible by optimal parallel work amount, and the new_m_dim is big enough
+    // In this case, each thread will execute the Snippets kernel 'batch_dim' times
+    if (m_dim % optimal_parallelism_work_amount == 0) {
+        const auto new_m_dim = m_dim / optimal_parallelism_work_amount;
+        const size_t min_kernel_m = 64;
+        if (new_m_dim >= min_kernel_m) {
+            splited.first = optimal_parallelism_work_amount;
+            splited.second = new_m_dim;
+            OPENVINO_ASSERT(splited.first * splited.second == m_dim, "Incorrect dimension M splitting!");
+            return splited;
+        }
+    }
+
```

Comment:
I think we need to return to this algorithm soon.
Because I'm not sure still that the current implementation covers our needs.
Just imagine that there is shape `[1,5,16384,64]` (SD) and `optimal_parallelism_work_amount = 18` (our workstation).
Ideal case #1 will be skipped because `18 / 5 = 3.6` - not integer value.
Ideal case #2 will be skipped too because `16384 / 18 = 910.(2)` - not integer value.
We will go to the next step and will get not optimal scheduling - I'd expect that in this case `new_m_dim` will be `32` or `64` (small) and `new_batch_dim` = `512` or `256` - yes, not all threads will process the same count of kernels. But there will a lot of kernels that we shouldn't notice this non-equality. This is my opinion and I'm not sure in 100% too. But I just change the thread count for SD - and Ideal Case #2 is broken now.

Reply:
Agree, I reflected this point in the 157339 ticket